datafusion/sql/src/planner.rs

```diff
@@ -3641,10 +3632,11 @@ mod tests {
         let sql = "SELECT id, age
                    FROM person
                    HAVING age > 100 AND age < 200";
-        let expected = "Projection: person.id, person.age\
-                        \n  Filter: person.age > Int64(100) AND person.age < Int64(200)\
-                        \n    TableScan: person";
-        quick_test(sql, expected);
+        let err = logical_plan(sql).expect_err("query should have failed");
```

Comment:
I recommend one test (that should also fail) where there is a GROUP clause but the having clause refers to an invalid column, such as

```sql
SELECT id, MAX(age)
FROM PERSON
GROUP BY id
-- first_name is a valid column but does not appear in the grouping output (group column nor aggregate)
HAVING first_name = 'M'
```

Reply:
Nice advice❤️
Added it.